src/dag/chunk.rs

```diff
@@ -38,16 +38,18 @@ impl Chunk {
         &(self.data.0[self.data.1..])
     }
 
-    // TODO: It would be nice to flatten the option down into an empty
-    // iterator for caller convenience, but could not find a zero-alloc
-    // way to do that.
-    pub fn refs(&self) -> Option<impl Iterator<Item = &str>> {
-        if let Some(buf) = self.meta() {
+    pub fn refs(&self) -> impl Iterator<Item = &str> {
+        let inner = if let Some(buf) = self.meta() {
```

Comment:
I think an even smaller way to do this is with `flat_map`. Something like:

```
self.meta().iter().flat_map(|buf| meta::get_root_as_meta(buf).refs()).iter().flat_map(|refs| refs.iter())
```
?